client/src/context/Data/DataState.js

```diff
@@ -0,0 +1,45 @@
+import React, { useReducer, useEffect } from "react";
+import DataContext from "./DataContext";
+import DataReducer from "./DataReducer";
+import { GET_DATA, SET_LOADING } from "../types";
+import axios from "axios";
+
+const DataState = (props) => {
+  const initialState = {
+    availableTopics: "",
+    loading: false,
+  };
+  
+  const [state, dispatch] = useReducer(DataReducer, initialState);
+
+  const setLoading = () => {
+    dispatch({ type: SET_LOADING });
+  };
+
+  useEffect(() => {
+    const getData = async () => {
+      setLoading();
+      try {
+        // console.log("Start loading data.");
+        const availableTopics = (await axios.get(`https://puxec9.deta.dev/keys`)).data;  
+        dispatch({ type: GET_DATA, payload: availableTopics });
+      } catch (err) {
+        console.error(err);
+      }
+    };
+    getData();
+  }, []);
```

Comment:
We shouldn't be using any useEffect's in this "state" component. Export `getData()` in the `value{{...}}` list below, and call the method on the page where data is needed (possibly in a `useEffect` on that page).  If you're trying to call `getData()` immediately when the website is used, then call `getData()` in `App.js`.